src/checkqueue.h

```diff
@@ -180,9 +186,7 @@ class CCheckQueueControl
     {
         // passed queue is supposed to be unused, or NULL
         if (pqueue != NULL) {
-            assert(pqueue->nTotal == pqueue->nIdle);
-            assert(pqueue->nTodo == 0);
-            assert(pqueue->fAllOk == true);
+            assert(pqueue->IsIdle());
```

Comment:
Can you avoid having an effectful statement inside an assert? We're requiring NDEBUG now, but that may not remain the case.